emulator/coverage_reported_runtime.go

```diff
@@ -0,0 +1,152 @@
+ * limitations under the License.
+ */
+
+package emulator
+
+import (
+	"github.com/onflow/cadence"
+	"github.com/onflow/cadence/runtime"
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/interpreter"
+	"github.com/onflow/cadence/runtime/sema"
+)
+
+type CoverageReportedRuntime struct {
+	runtime.Runtime
+	runtime.Environment
+	*runtime.CoverageReport
+}
+
+func (crr CoverageReportedRuntime) NewScriptExecutor(
+	script runtime.Script,
+	context runtime.Context,
+) runtime.Executor {
+	context.CoverageReport = crr.CoverageReport
+	return crr.Runtime.NewScriptExecutor(script, context)
+}
+
+func (crr CoverageReportedRuntime) ExecuteScript(
+	script runtime.Script,
+	context runtime.Context,
+) (
+	cadence.Value,
+	error,
+) {
+	context.CoverageReport = crr.CoverageReport
+	return crr.Runtime.ExecuteScript(script, context)
+}
+
+func (crr CoverageReportedRuntime) NewTransactionExecutor(
+	script runtime.Script,
+	context runtime.Context,
+) runtime.Executor {
+	context.CoverageReport = crr.CoverageReport
+	return crr.Runtime.NewTransactionExecutor(script, context)
+}
+
+func (crr CoverageReportedRuntime) ExecuteTransaction(
+	script runtime.Script,
+	context runtime.Context,
+) error {
+	context.CoverageReport = crr.CoverageReport
+	return crr.Runtime.ExecuteTransaction(script, context)
+}
+
+func (crr CoverageReportedRuntime) NewContractFunctionExecutor(
+	contractLocation common.AddressLocation,
+	functionName string,
+	arguments []cadence.Value,
+	argumentTypes []sema.Type,
+	context runtime.Context,
+) runtime.Executor {
+	context.CoverageReport = crr.CoverageReport
+	return crr.Runtime.NewContractFunctionExecutor(
+		contractLocation,
+		functionName,
+		arguments,
+		argumentTypes,
+		context,
+	)
+}
+
+func (crr CoverageReportedRuntime) InvokeContractFunction(
+	contractLocation common.AddressLocation,
+	functionName string,
+	arguments []cadence.Value,
+	argumentTypes []sema.Type,
+	context runtime.Context,
+) (
+	cadence.Value,
+	error,
+) {
+	context.CoverageReport = crr.CoverageReport
+	return crr.Runtime.InvokeContractFunction(
+		contractLocation,
+		functionName,
+		arguments,
+		argumentTypes,
+		context,
+	)
+}
+
+func (crr CoverageReportedRuntime) ParseAndCheckProgram(
+	source []byte,
+	context runtime.Context,
+) (
+	*interpreter.Program,
+	error,
+) {
+	context.CoverageReport = crr.CoverageReport
+	return crr.Runtime.ParseAndCheckProgram(source, context)
+}
+
+func (crr CoverageReportedRuntime) ReadStored(
+	address common.Address,
+	path cadence.Path,
+	context runtime.Context,
+) (
+	cadence.Value,
+	error,
+) {
+	context.CoverageReport = crr.CoverageReport
+	return crr.Runtime.ReadStored(address, path, context)
+}
+
+func (crr CoverageReportedRuntime) ReadLinked(
+	address common.Address,
+	path cadence.Path,
+	context runtime.Context,
+) (
+	cadence.Value,
+	error,
+) {
+	context.CoverageReport = crr.CoverageReport
+	//nolint:staticcheck
+	return crr.Runtime.ReadLinked(address, path, context)
```

Comment:
What linting issue does this line have?

Reply:
About `ReadLinked` being deprecated. Same as here: https://github.com/onflow/flow-go/blob/fc98c1cabbb04c5197b03911aef542099c55ae0b/fvm/runtime/wrapped_cadence_runtime.go#L71

Reply:
See https://github.com/onflow/cadence/blob/1044a9b9155c348a882b66ee7b31ca2fe719c40c/runtime/runtime.go#L150-L152